### PR TITLE
clh: Fix version check

### PR DESCRIPTION
**Pull request by jcvenegas**
Add support for new version format from clh.

Fixes: #2489

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by jcvenegas** [comment]
@likebreath @sameo

File: virtcontainers/clh.go [hunk 1]
```diff
@@ -675,6 +675,9 @@ func (clh *cloudHypervisor) getAvailableVersion() error {
 
 	}
 
+	// Remove 'v' prefix if has one
+	clh.Logger().Error("DEBUG :", versionSplit[0])
```
Comment (grahamwhaley) [inline virtcontainers/clh.go]: The combination of `Error` and `DEBUG` here doesn't feel right...

File: virtcontainers/clh.go [hunk 2]
```diff
@@ -676,7 +676,6 @@ func (clh *cloudHypervisor) getAvailableVersion() error {
 	}
 
 	// Remove 'v' prefix if has one
-	clh.Logger().Error("DEBUG :", versionSplit[0])
```
Comment (grahamwhaley) [inline virtcontainers/clh.go]: Hah, and in the next patch you dropped it .... probably better if you clean that up to not have it in the first place.

File: virtcontainers/clh.go [hunk 3]
```diff
@@ -685,10 +687,15 @@ func (clh *cloudHypervisor) getAvailableVersion() error {
 		return err
 
 	}
-	revision, err := strconv.ParseUint(versionSplit[2], 10, 64)
+
+	// revision could have aditional commit information separated by '-'
+	revisionSplit := strings.SplitN(versionSplit[2], "-", -1)
+	if len(revisionSplit) < 1 {
+		return errors.Errorf("Failed parse cloud-hypervisor revision %s", versionSplit[2])
```
Comment (jodh-intel) [inline virtcontainers/clh.go]: Might be worth logging `versionSplit` rather than `versionSplit[2]`?

**Pull request merged**